paimon-common/src/main/java/org/apache/paimon/types/DataTypeJsonParser.java

```diff
@@ -66,8 +73,10 @@ public static DataType parseDataType(JsonNode json) {
                 JsonNode fieldArray = json.get("fields");
                 Iterator<JsonNode> iterator = fieldArray.elements();
                 List<DataField> fields = new ArrayList<>(fieldArray.size());
+                int id = 0;
                 while (iterator.hasNext()) {
-                    fields.add(parseDataField(iterator.next()));
+                    fields.add(parseDataField(iterator.next(), id));
```

Comment:
Not work for nested types?

Reply:
Could you provide an example, not sure which nested types